climada/hazard/wildfire.py

```diff
@@ -0,0 +1,1165 @@
+from climada.hazard.base import Hazard
+from climada.hazard.tag import Tag as TagHazard
+from climada.util.constants import ONE_LAT_KM, DEF_CRS
+from climada.util.dates_times import str_to_date
+
+from climada.util.alpha_shape import alpha_shape, plot_polygon
+warnings.simplefilter(action='ignore', category=FutureWarning)
+
+LOGGER = logging.getLogger(__name__)
+
+HAZ_TYPE = 'WF'
+""" Hazard type acronym for Wild Fire, might be changed to WFseason or WFsingle """
+
+CLEAN_THRESH = 30
+""" Minimal confidence value for the data from MODIS instrument to be use as input"""
+
+RES_DATA = 1.0
+""" Resolution of the data if no data origin provided (km) """
+
+BLURR_STEPS = 4
+""" steps with exponential decay for fire propagation matrix """
+
+class WildFire(Hazard):
+
+    """Contains wild fires.
+
+    Wildfires comprise the challange that the definition of an event is unclear.
+    Reporting standards vary accross regions and over time. Hence, to have
+    consistency, we consider an event as a whole fire season. This allows
+    consistent risk assessment across the global and over time. Events that
+    refer to a fire season have the tag 'WFseason'.
+
+    In order to perform concrete case studies or calibrate impact functions,
+    events can be displayed as single fires. In that case they have the tag
+    'WFsingle'.
+
+
+    Attributes:
+        date_end ((np.array): integer date corresponding to the proleptic
+            Gregorian ordinal, where January 1 of year 1 has ordinal 1
+            (ordinal format of datetime library))
+
+        n_fires (np.array): number of single fires in a fire season
+
+    """
+
+    days_thres_firms = 2
+    """ Minimum number of days to consider different fires """
+
+    clus_thres_firms = 15
+    """ Clustering factor which multiplies instrument resolution """
+
+    remove_minor_fires_firms = True
+    """ removes FIRMS fires below defined theshold of entries"""
+
+    minor_fire_thres_firms = 3
+    """ number of FIRMS entries required to be considered a fire """
+
+    prop_proba = 0.21
+    """ probability of fire propagation """
+
+    max_it_propa = 500000
+    """ maximum propgation iterations for a probabilistic fire """
+
+    def __init__(self, pool=None):
+        """Empty constructor. """
+        Hazard.__init__(self, HAZ_TYPE)
+        if pool:
+            self.pool = pool
+            LOGGER.info('Using %s CPUs.', self.pool.ncpus)
+        else:
+            self.pool = None
+
+    def set_hist_fire_FIRMS(self, csv_firms, centr_res_factor=1, centroids=None):
+        """ Parse FIRMS data and generate historical fires by temporal and spatial
+        clustering.
+
+        Parameters:
+            csv_firms: csv file of the FIRMS data (https://firms.modaps.eosdis.nasa.gov/download/)
+                or pd.DataFrame of FIRMS data
+            centr_res_factor (int, optional): resolution factor with respect to
+                the satellite data to use for centroids creation. Default: 1
```

Comment:
This is not very clear. Do you mean that you create a raster with a given resolution? What is the unit? Also, what happens if one specifies both `centr_res_factor` and `centroids`?

Reply:
I've adjusted the docsting (commit 91b6f19f)

centr_res_factor : float, optional, default=1
                resolution factor with respect to the satellite data to use
                for centroids creation. Hence, if MODIS data (1 km res) is
                used and centr_res_factor is set to 0.2, the grid spacing of
                the generated centroids will equal 5 km (=1/0.2). If centroids
                are defined, this parameter has no effect.